src/transformers/modeling_utils.py

```diff
@@ -2126,6 +2116,24 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                     "Using `low_cpu_mem_usage=True` or a `device_map` requires Accelerate: `pip install accelerate`"
                 )
 
+        quantization_config_kwargs = {
+            k: v for k, v in kwargs.items() if k in inspect.signature(BitsAndBytesConfig).parameters
+        }
```

Comment:
It would be easier to have `BitsAndBytesConfig` have a `from_dict` method with `return_unused_kwargs`, like `PretrainedConfig`. We can then do
```
quantization_config, kwargs = BitsAndBytesConfig(**kwargs)
```

Reply:
Agreed! 
I think we stil need to add: 
```python

quantization_config_kwargs = {
                k: v for k, v in kwargs.items() if k in inspect.signature(BitsAndBytesConfig).parameters
            }
```
 to check if a user has passed `quantization_config` together with some other associated kwargs inside `from_pretrained`